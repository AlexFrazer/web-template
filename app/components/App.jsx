import React, { Component } from 'react';

import styles from 'app/styles/app.scss';

export default class App extends Component {
  static propTypes = {
  };

  state = { disabled: false };

  toggleButton = open => this.setState({ disabled: !open });

  render() {
    return (
      <div className={styles.container}>
        <div className={styles.content}>
          Test
        </div>
      </div>
    );
  }
}
