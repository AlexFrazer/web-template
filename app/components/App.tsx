import * as React from 'react';
import * as styles from 'app/styles/app.scss';
import Counter from 'app/components/counter/Counter';

export default function app() {
  return (
    <div className={styles.container}>
      <Counter title="Default Counter" />
      <Counter title="Initialized Counter" initial={1} />
    </div>
  );
}
