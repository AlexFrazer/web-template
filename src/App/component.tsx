import * as React from 'react';
import { Switch, Route, BrowserRouter } from 'react-router-dom';
import { BounceLoader } from 'react-spinners';
import Loadable from 'react-loadable';

export default function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route
          exact
          path="/about"
          component={Loadable({
            loader: () => import('./About' /* webpackChunkName: "About" */),
            loading: () => <BounceLoader loading />,
          })}
        />
        <Route
          path="/"
          component={Loadable({
            loader: () => import('./Counter' /* webpackChunkName: "Counter" */),
            loading: () => <BounceLoader loading />,
          })}
        />
      </Switch>
    </BrowserRouter>
  );
}
