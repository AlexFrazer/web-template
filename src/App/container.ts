import { hot } from 'react-hot-loader';
import component from './component';

export default hot(module)(component);