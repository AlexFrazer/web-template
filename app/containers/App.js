import { connect } from 'react-redux';

import App from 'app/components/App';

const mapStateToProps = state => ({
  isOpen: state.sidebar.isOpen,
});

export default connect(mapStateToProps)(App);
