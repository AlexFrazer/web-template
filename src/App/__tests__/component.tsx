import * as React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import App from '../';

describe('<App />', () => {
  it('Should render the app.', () => {
    expect(toJson(shallow(<App />))).toMatchSnapshot();
  });
});
