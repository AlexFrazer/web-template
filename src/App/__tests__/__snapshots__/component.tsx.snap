// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render the app. 1`] = `
<React.Suspense
  fallback={<div />}
>
  <ThemeProvider
    theme={
      Object {
        "palette": Object {
          "primary": "#1a1a1a",
        },
      }
    }
  >
    <BrowserRouter>
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/todos"
        />
        <Redirect
          push={false}
          to="/"
        />
      </Switch>
    </BrowserRouter>
  </ThemeProvider>
</React.Suspense>
`;
